pkg/filters/binary.go

```diff
@@ -0,0 +1,203 @@
+package filters
+
+import (
+	"encoding/binary"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+	"github.com/aquasecurity/tracee/pkg/capabilities"
+	"github.com/aquasecurity/tracee/pkg/utils/proc"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+)
+
+type nsBinary struct {
+	mntNS uint32
+	path  string
+}
+
+type BinaryFilter struct {
+	equal    map[nsBinary]bool
+	notEqual map[nsBinary]bool
+	enabled  bool
+}
+
+func getHostMntNS() (uint32, error) {
+	var ns int
+	var err error
+
+	err = capabilities.GetInstance().Requested(func() error {
+		ns, err = proc.GetProcNS(1, "mnt")
+		return err
+	},
+		cap.DAC_READ_SEARCH,
+		cap.SYS_PTRACE,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	return uint32(ns), nil
+}
+
+func NewBinaryFilter() *BinaryFilter {
+	return &BinaryFilter{
+		equal:    map[nsBinary]bool{},
+		notEqual: map[nsBinary]bool{},
+	}
+}
+
+func (f *BinaryFilter) Parse(operatorAndValues string) error {
+	if len(operatorAndValues) < 2 {
+		return InvalidExpression(operatorAndValues)
+	}
+	valuesString := string(operatorAndValues[1:])
+	operatorString := string(operatorAndValues[0])
+
+	if operatorString == "!" {
+		if len(operatorAndValues) < 3 {
+			return InvalidExpression(operatorAndValues)
+		}
+		operatorString = operatorAndValues[0:2]
+		valuesString = operatorAndValues[2:]
+	}
+
+	values := strings.Split(valuesString, ",")
+
+	var hostMntNS uint32
+	var err error
+
+	for _, val := range values {
+		mntAndPath := strings.Split(val, ":")
+		var bin nsBinary
+		if len(mntAndPath) == 1 {
+			bin.path = val
+		} else if len(mntAndPath) == 2 {
+			bin.path = mntAndPath[1]
+			if mntAndPath[0] == "host" {
+				if hostMntNS == 0 {
```

Comment:
nit: We can probably cache this for the first time then reuse later and short circuit if necessary.

Reply:
This is exactly what I'm already doing here.
hostMntNS is defined above, and once it becomes non-zero, it is considered cached

Reply:
No I mean you can do it above the parsing loop. Store it in the struct like before, do the check then fail even earlier. This is mostly relevant if we ever want to do runtime parsing of this filter.
Maybe even do it in the constructor and add an error value there (but it will make the filter initialization code slightly more annoying(.

Reply:
No I mean you can do it above the parsing loop. Store it in the struct like before, do the check then fail even earlier. This is mostly relevant if we ever want to do runtime parsing of this filter.
Maybe even do it in the constructor and add an error value there (but it will make the filter initialization code slightly more annoying).

Reply:
So other than failing earlier it has no real benefit since caching is already imlpemented here.
I think it is ok to continue as is.